doc/develop/test/twister.rst

```diff
@@ -408,6 +408,24 @@ platform_allow: <list of platforms>
     constraints, this option should only be used if the test or sample can
     only be run on the allowed platform and nothing else.
 
+    .. warning::
```

Comment:
I think this (the description of `platform_allow` field) is a wrong place to describe the scope selection quirks. IMO a new section has to be added that will contain such information (e.g. `Scope selection`).

Reply:
I disagree, it will just scatter the documentation. What's describe there is the behavior of that specific option, when user will look for it they will immediately see it and it will avoid them some back and forth on the documentation.